src/cli/commands/wallet/list.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+//! Wallet new cli handler
+
+use async_trait::async_trait;
+use clap::Args;
+use std::fmt::Debug;
+
+use crate::cli::commands::get_ipc_agent_url;
+use crate::cli::{CommandLineHandler, GlobalArguments};
+use crate::config::json_rpc_methods;
+use crate::jsonrpc::{JsonRpcClient, JsonRpcClientImpl};
+use crate::server::wallet::list::{WalletListParams, WalletListResponse};
+
+pub(crate) struct WalletList;
+
+#[async_trait]
+impl CommandLineHandler for WalletList {
+    type Arguments = WalletListArgs;
+
+    async fn handle(global: &GlobalArguments, arguments: &Self::Arguments) -> anyhow::Result<()> {
+        log::debug!("list wallets with args: {:?}", arguments);
+
+        let url = get_ipc_agent_url(&arguments.ipc_agent_url, global)?;
+        let json_rpc_client = JsonRpcClientImpl::new(url, None);
+
+        let params = WalletListParams {
+            subnet: arguments.subnet.clone(),
+        };
+
+        let addrs = json_rpc_client
+            .request::<WalletListResponse>(
+                json_rpc_methods::WALLET_LIST,
+                serde_json::to_value(params)?,
+            )
+            .await?;
+
+        log::info!("wallets in subnet {:} are {:?}", arguments.subnet, addrs);
```

Comment:
We are still not showing the balance for each address here, right? I would love to include this in the docs (I think it could really improve the UX)